Hermes/helper.py

```diff
@@ -231,6 +233,33 @@ def delete_task_from_mem(task_id):
     del TASK_STATUS[task_id]
   TASK_STATUS_LOCK.release()
 
+def get_all_stats():
+  """ Collects platform stats from all deployment nodes.
+
+  Returns:
+    A dictionary containing all the monitoring stats, if all nodes are
+    accessible. {"success": False, "error": message} otherwise.
+  """
+  all_stats = {}
+
+  secret = appscale_info.get_secret()
+  logging.debug("Retrieved deployment secret: {}".format(secret))
+  for ip in appscale_info.get_all_ips():
+    appcontroller_endpoint = "https://{}:{}".format(ip,
+      hermes_constants.APPCONTROLLER_PORT)
+    logging.debug("Connecting to AC at: {}".format(appcontroller_endpoint))
+    # Do a SOAP call to the AppController on that IP to get stats.
+    server = SOAPpy.SOAPProxy(appcontroller_endpoint)
+    try:
+      all_stats[ip] = server.get_all_stats(secret)
+    except SOAPpy.SOAPException as soap_exception:
+      logging.exception("Exception while performing SOAP call to "
+        "{}".format(appcontroller_endpoint))
+      logging.exception(soap_exception)
+      return {"success": False, "error": "Node {} inaccessible".format(ip)}
```

Comment:
It could return the stats for the accessible nodes and then an error for any nodes that are unreachable. Thanks for bringing this up.